source/Octopus.Client/Util/ParameterHelper.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Octopus.Client.Exceptions;
+using Octopus.Client.Repositories.Async;
+
+namespace Octopus.Client.Util
+{
+    static class ParameterHelper
+    {
+        public static Dictionary<string, object> CombineParameters(Dictionary<string, object> currentAdditionalQueryParameters, object userProvidedParameters)
+        {
+            if (userProvidedParameters == null)
+                userProvidedParameters = new { };
+            var resultDictionary = GetParameters(userProvidedParameters);
+
+            // Value from the current repository(currentAdditionalQueryParameters) overrides the one from user (userProvidedParameters)
+            currentAdditionalQueryParameters.ToList().ForEach(x => resultDictionary[x.Key] = x.Value);
+            return new Dictionary<string, object>(resultDictionary, StringComparer.OrdinalIgnoreCase);
+        }
+
+        public static void ValidateSpaceParameters(SpaceQueryParameters currentSpaceQueryParameters, SpaceQueryParameters newSpaceQueryParameters)
```

Comment:
This feels like it should be in `MixedScopeBaseRepository`. I get why you have pulled it out though (because we have duplicated code between the sync and async repos). We have a lot of duplicate code actually, so this is not a new problem :(. The thing I don't like about this implementation is that it makes an assumption that `currentSpaceQueryParameters` can be null, which feels like an implementation detail of `MixedScopeBaseRepository`. I think if we can pull the null check up, that would be much better.